pkg/pods/namespace.go

```diff
@@ -14,44 +13,14 @@ var (
 
 const (
 	// The corresponding environment variable is configured to contain pod namespace by sensor YAML/helm file using
-	// the Kubernetes Downward API.
+	// the Kubernetes Downward API, see
+	// https://github.com/kubernetes/kubernetes/blob/release-1.0/docs/user-guide/downward-api.md
 	nsEnvVar = "POD_NAMESPACE"
 )
 
-type HeuristicType int
-
-const (
-	// NoSATokenNamespace is the default option. It does *not* read the SAToken
-	// file. For more information on why this mechanism is the default, see
-	//     https://issues.redhat.com/browse/ROX-12349
-	NoSATokenNamespace HeuristicType = iota
-	// ConsiderSATokenNamespace switches the legacy mechanism which first reads the
-	// SAToken file namespace. While this mechanism works _most_ of the time, on
-	// our CI runners we saw the service account namespace set to `ci-op-<HASH`,
-	// see this for more information
-	//     https://github.com/openshift/ci-operator/blob/master/TEMPLATES.md
-	//
-	// This options is kept for now to avoid unexpected side effects but we aim
-	// to remove it in the future once we ensure that the POD_NAMESPACE env var
-	// is set correctly via the Kubernetes Downward API for all our pods, see
-	//     https://github.com/kubernetes/kubernetes/blob/release-1.0/docs/user-guide/downward-api.md
-	ConsiderSATokenNamespace
-)
-
-// GetPodNamespace is a heuristic to determine in which namespace a given Pod runs.
-func GetPodNamespace(heuristic HeuristicType) string {
-	var sensorNamespace string
-	if heuristic == ConsiderSATokenNamespace {
-		var err error
-		sensorNamespace, err = satoken.LoadNamespaceFromFile()
-		if err != nil {
-			log.Errorf("Failed to determine namespace from service account token file: %s", err)
-		}
-	}
-
-	if sensorNamespace == "" {
-		sensorNamespace = os.Getenv(nsEnvVar)
-	}
+// GetPodNamespace is a heuristic to determine in what namespace a given Pod runs.
```

Comment:
Let's leave some breadcrumbs for posterity, who might be us debugging an issue in a year. How about:

```suggestion
// GetPodNamespace is a heuristic to determine in what namespace a given pod
// runs. It relies on the POD_NAMESPACE env var being correctly set. For more
// information on why this mechanism is the default, see
//     https://issues.redhat.com/browse/ROX-12349
//
// We don't read the SAToken file anymore because it sometimes lies, see
//     https://github.com/openshift/ci-operator/blob/master/TEMPLATES.md
```

Reply:
Sure, I removed a lot of those after getting rid of SA namespace option but I guess it can't hurt to be verbose here.